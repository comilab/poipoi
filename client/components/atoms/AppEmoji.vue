<template lang="pug">
.emoji-wrapper.inline-block(
  v-html="img"
)
</template>

<script lang="ts">
import { defineComponent, computed } from 'nuxt-composition-api'
import Twemoji from 'twemoji'

export default defineComponent({
  props: {
    text: {
      type: String,
      required: true
    }
  },
  setup (props) {
    const img = computed(() => Twemoji.parse(props.text))

    return {
      img
    }
  }
})
</script>

<style lang="postcss" scoped>
.emoji-wrapper {
  line-height: 0;

  & >>> img {
    @apply inline-block;
  }
}

</style>
