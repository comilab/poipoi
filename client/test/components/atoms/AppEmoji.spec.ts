import { render, RenderOptions } from '@testing-library/vue'
import { mocked } from 'ts-jest/utils'
import Twemoji from 'twemoji'

import AppEmoji from '~/components/atoms/AppEmoji.vue'

jest.mock('twemoji')

describe('components/atoms/AppEmoji', () => {
  let options: RenderOptions<AppEmoji>

  beforeEach(() => {
    options = {
      props: {
        text: '🍣'
      }
    }

    mocked(Twemoji.parse).mockImplementation((emoji: string|HTMLElement) => {
      return `<img alt="${emoji}" />`
    })
  })

  test('renders correctly', () => {
    const { container, getByAltText } = render(AppEmoji, options)

    expect(container).toMatchSnapshot()
    getByAltText('🍣')
    expect(Twemoji.parse).toBeCalledWith('🍣')
  })
})
