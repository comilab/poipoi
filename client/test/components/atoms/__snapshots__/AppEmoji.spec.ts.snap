// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/AppEmoji renders correctly 1`] = `
<div>
  <div
    class="emoji-wrapper inline-block"
  >
    <img
      alt="🍣"
    />
  </div>
</div>
`;
